### PR TITLE
Add Subtitulamos provider integration

### PR DESCRIPTION
**Pull request by Nyaran**
Added Subtitulamos provider

This PR is partially based and replaces #826 (skipping the provider "tusubtitulo" as is not working anymore)

~~Also I added a way to run the cli from source code ([subliminal/cli.py#L727-L728](https://github.com/Nyaran/subliminal/blob/6699c62f221d2e6392f86fed008265a01d5bc316/subliminal/cli.py#L727-L728)) , as is helpful to debug new additions (if you don't like it, tell me and I'll remove it)~~

Pending:

- [x] Add tests
- [x] Changelog entry
- [x] CI passes (except docs)

### CODE REVIEW

File: subliminal/cli.py [hunk 1]
```diff
@@ -722,3 +722,7 @@ def download(
 
     if verbose == 0:
         click.echo(f"Downloaded {plural(total_subtitles, 'subtitle')}")
+
+
+if __name__ == '__main__':
+    subliminal()
```
Comment (ptrcnull) [inline subliminal/cli.py:728]: this should be in a separate commit, if anything

Comment (ptrcnull) [reply subliminal/cli.py:728]: also, if this was just a temporary launcher thingy for development, you can achieve the same with `python3 -m subliminal`

Comment (Nyaran) [reply subliminal/cli.py:728]: I tried that, and I get " No module named subliminal.__main__; 'subliminal' is a package and cannot be directly executed" (I'm running from the virtual env, after installing dependencies)

Anyway, no problem, before I finish this PR I will reverse the change

**Comment by getzze** [comment]
Thanks for updating this provider!

Don't worry too much about mypy (or if you need help let me know). It's better if coverage is high ;)

File: subliminal/cli.py (hunk 1 above)
Comment (getzze) [reply subliminal/cli.py]: You can install subliminal in editable mode with `pip install '.[dev]' -e`, even in a virtual env. Like that you can install subliminal and run it's CLI from the terminal, and at the same time modify the files and see the modification in the installed package.

File: subliminal/providers/subtitulamos.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+            language = Language.fromsubtitulamos(lang_name_element.get_text().strip())
+
+            hearing_impaired = False
+
+            # modify spanish latino subtitle language to only spanish and set hearing_impaired = True
+            # because if exists spanish and spanish latino subtitle for the same episode, the score will be
+            # higher with spanish subtitle. Spanish subtitle takes priority.
+            if language == Language('spa', 'MX'):
+                language = Language('spa')
+                hearing_impaired = True
```
Comment (getzze) [inline subliminal/providers/subtitulamos.py:238]: this is a hack that will stop working with changes to come to how `hearing_impaired` and `foreign_only`/`forced` subtitles are sorted. See #1175 for the first step.

Why do you need to transform a american-spanish subtitle to a european-spanish subtitle? It can be specified by having both languages (or only one) in the command line, no?

Comment (Nyaran) [reply subliminal/providers/subtitulamos.py:238]: I'm not really sure, I thought the same when I saw that on the original proposal, but I decided to keep it in case there was something I missed. Let me perform some test and make the commit removing the block.

Comment (Nyaran) [reply subliminal/providers/subtitulamos.py]: Done, and after reviewing it I improved this to support all countries with Spanish language (all of them under the name "Español (Latinoamérica)" on the provider.

File: subliminal/providers/subtitulamos.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+                or (lang_name_element := sub.find_previous('div', class_='language-name')) is None
+                or (version_container := sub.find_previous('div', class_='version-container')) is None
+                or not isinstance(version_container, Tag)
+                or (release_group_element := version_container.select('.version-container .text.spaced')) is None
+            ):
+                continue
+
+            language = Language.fromsubtitulamos(lang_name_element.get_text().strip())
+
+            hearing_impaired = False
```
Comment (getzze) [inline subliminal/providers/subtitulamos.py:259]: Do we need to force `hearing_impaired = False`, meaning we are sure they ARE NOT hearing impaired? The default, `None`, means we don't know.

File: subliminal/providers/subtitulamos.py (hunk 2 above)
Comment (getzze) [reply subliminal/providers/subtitulamos.py]: Thanks!

File: subliminal/providers/subtitulamos.py (hunk 3 above)
Comment (Nyaran) [reply subliminal/providers/subtitulamos.py]: I think subtitulamos doesn't have hearing impaired subtitles, but I'm not sure, so... line removed

Comment (getzze) [reply subliminal/providers/subtitulamos.py]: ok, thanks

**Comment by getzze** [comment]
Great job!
